ark/utils/load_utils_test.py

```diff
@@ -172,23 +172,99 @@ def test_load_imgs_from_tree():
 
+        # invalid list of files is provided
+        with pytest.raises(ValueError):
+            loaded_xr = \
+                load_utils.load_imgs_from_dir(temp_dir, files=fovs + ['not_an_image'],
+                                              delimiter='_', dtype=np.float32)
+        with pytest.raises(ValueError):
+            loaded_xr = \
+                load_utils.load_imgs_from_dir(temp_dir, files=['not_an_image'],
+                                              delimiter='_', dtype=np.float32)
+
         # check default loading
         loaded_xr = \
-            load_utils.load_imgs_from_dir(temp_dir, delimiter='_', dtype=np.float32)
+            load_utils.load_imgs_from_dir(temp_dir, delimiter='_',
+                                          xr_dim_name='compartments', dtype=np.float32)
 
         assert loaded_xr.equals(data_xr)
 
         # test swap int16 -> float
         with pytest.warns(UserWarning):
             loaded_xr = \
-                load_utils.load_imgs_from_dir(temp_dir, delimiter='_', dtype="int16")
+                load_utils.load_imgs_from_dir(temp_dir, delimiter='_',
+                                              xr_dim_name='compartments', dtype="int16")
 
             assert loaded_xr.equals(data_xr)
             assert np.issubdtype(loaded_xr.dtype, np.floating)
+
+    # test multitiff input
+    with tempfile.TemporaryDirectory() as temp_dir:
+        fovs, channels = test_utils.gen_fov_chan_names(num_fovs=2, num_chans=3, use_delimiter=True)
+
+        filelocs, data_xr = test_utils.create_paired_xarray_fovs(
+            temp_dir, fovs, channels, img_shape=(10, 10), mode='reverse_multitiff', delimiter='_',
+            fills=True, dtype=np.float32
+        )
+
+        fovnames = [f'{fov}.tiff' for fov in fovs]
+
+        # test all channels loading w/ specified file
+        loaded_xr = \
+            load_utils.load_imgs_from_dir(temp_dir, files=[fovnames[-1]], xr_dim_name='channels',
+                                          delimiter='_', dtype=np.float32)
+
+        assert loaded_xr.equals(data_xr.loc[[fovs[-1]], :, :, :])
+
+        # test all channels w/ unspecified files + delimiter agnosticism
+        loaded_xr = load_utils.load_imgs_from_dir(temp_dir,
+                                                  files=None,
+                                                  channel_indices=None,
+                                                  xr_dim_name='channels',
+                                                  delimiter='_')
+
+        assert loaded_xr.equals(data_xr)
+
+        # test channels_first input
+        fovs, channels = test_utils.gen_fov_chan_names(num_fovs=2, num_chans=5, use_delimiter=True)
+
+        _, data_xr = test_utils.create_paired_xarray_fovs(
+            temp_dir, fovs, channels, img_shape=(10, 10), mode='multitiff', delimiter='_',
+            fills=True, dtype=np.float32, channels_first=True
```

Comment:
It looks like there isn't an assert statement here checking that this is correct

Reply:
There is an assert statement in line 261 for this input, or did you mean to validate that the input data is created correctly before calling `load_imgs_from_dir`?

Reply:
Ah my mistake, I didn't realize the one on 270 was testing the same thing with all fovs.